networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/BeaconChainMethods.java

```diff
@@ -290,6 +308,50 @@ private static Eth2RpcMethod<GoodbyeMessage, GoodbyeMessage> createGoodBye(
     }
   }
 
+  private static Optional<
+          Eth2RpcMethod<
+              BeaconBlockAndBlobsSidecarByRootRequestMessage, SignedBeaconBlockAndBlobsSidecar>>
+      createBeaconBlockAndBlobsSidecarByRoot(
+          final Spec spec,
+          final MetricsSystem metricsSystem,
+          final AsyncRunner asyncRunner,
+          final RecentChainData recentChainData,
+          final PeerLookup peerLookup,
+          final RpcEncoding rpcEncoding) {
+    if (!spec.isMilestoneSupported(SpecMilestone.EIP4844)) {
+      return Optional.empty();
+    }
+
+    final BeaconBlockAndBlobsSidecarByRootRequestMessageSchema requestType =
+        BeaconBlockAndBlobsSidecarByRootRequestMessage.SSZ_SCHEMA;
+
+    final SignedBeaconBlockAndBlobsSidecarSchema beaconBlockAndBlobsSidecarSchema =
+        spec.forMilestone(SpecMilestone.EIP4844)
+            .getSchemaDefinitions()
+            .toVersionEip4844()
+            .orElseThrow()
```

Comment:
Should be able to use:
```suggestion
        SchemaDefinitionsEip4844.required(
            spec.forMilestone(SpecMilestone.EIP4844).getSchemaDefinitions())
```

It will just give a better error message if it ever fails. :)

Reply:
thanks